### PR TITLE
Mule 17418 2

### PR DESCRIPTION
**Pull request by Bardammu**

### CODE REVIEW

File: src/main/java/org/mule/extension/db/internal/resolver/param/StoredProcedureParamTypeResolver.java [hunk 1]
```diff
@@ -109,20 +115,18 @@ public StoredProcedureParamTypeResolver(DbTypeManager dbTypeManager) {
 
       int typeId = procedureColumns.getInt(TYPE_ID_COLUMN_INDEX);
       String typeName = procedureColumns.getString(TYPE_NAME_COLUMN_INDEX);
+      String parameterName = procedureColumns.getString(PARAM_NAME_COLUMN_INDEX);
```
Comment (ndinu) [inline src/main/java/org/mule/extension/db/internal/resolver/param/StoredProcedureParamTypeResolver.java]: Would this affect performance?

File: src/main/java/org/mule/extension/db/internal/resolver/param/StoredProcedureParamTypeResolver.java [hunk 2]
```diff
@@ -109,20 +115,18 @@ public StoredProcedureParamTypeResolver(DbTypeManager dbTypeManager) {
...
+                   storedProcedureName, parameterName, position, typeId, typeName);
 
       DbType dbType = null;
       try {
         // TODO - MULE-15241 : Fix how DB Connector chooses ResolvedTypes
-        if (TABLE_TYPE_NAME.equals(typeName)) {
-          String procedureName = procedureColumns.getString(3);
-          String argumentName = procedureColumns.getString(4);
-          String owner = procedureColumns.getString(2);
+        if (typeName.equals(TABLE_TYPE_NAME)) {
```
Comment (ndinu) [inline src/main/java/org/mule/extension/db/internal/resolver/param/StoredProcedureParamTypeResolver.java]: can typeName be null?

File: src/main/java/org/mule/extension/db/internal/domain/connection/DbConnection.java [hunk 3]
```diff
@@ -136,6 +138,31 @@ default Struct createStruct(String typeName, Object[] values) throws SQLExceptio
...
+   * Some Databases assign a different meaning to each parameter. You should override this method when needed.
+   *
+   * @param storedProcedureName the stored procedure name
+   * @param storedProcedureOwner the owner of the stored procedure
+   * @param storedProcedureParentOwner the owner of the owner of the stored procedure
+   * @param catalogName the catalog name where the stored procedure is defined
+   * @return <code>ResultSet</code> - each row describes a stored procedure parameter or column
+   * @throws SQLException if a database access error occurs
+   */
+  default ResultSet getProcedureColumns(String storedProcedureName, String storedProcedureOwner,
```
Comment (ndinu) [inline src/main/java/org/mule/extension/db/internal/domain/connection/DbConnection.java]: Maybe have no default and implement it in DefaultConnection and implement it in XADbConnection?

File: src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnection.java [hunk 4]
```diff
@@ -176,4 +184,57 @@ protected void resolveLobs(String typeName, Object[] attributes, StructAndArrayT
...
+     * procedure name.
+     */
+    DatabaseMetaData dbMetaData = getJdbcConnection().getMetaData();
+
+    String connectionSchema;
+    try {
+      connectionSchema = getJdbcConnection().getSchema();
+    } catch (Throwable t) {
+      LOGGER
+          .warn("You are using a not supported jdbc driver version. Consider to upgrade to a new version to guarantee a better performance.");
```
[the remark this answers is not in the document]
Comment (Bardammu) [reply src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnection.java]: Oracle drivers compiled with 1.6 java

**Pull request merged**